kover-gradle-plugin/docs/index.md

```diff
@@ -36,11 +42,677 @@ Gradle plugin that measures the coverage for tests running on the JVM and genera
+```kotlin
+kover {
+    reports {
+        total {
+            verify {
+                // add new verification rule
+                rule {
+                    // specify coverage bounds for the rule
+                }
+            }
+        }
+    }
+}
+```
+
+You can read more about creating verification rules in the [corresponding section](#verification).
+
+
+#### Configuring Reports
+To configure of the report generation for `custom` report variant, it is necessary to specify the following in the build script file of the merging module (and only in it):
+
+```kotlin
+kover {
+    reports {
+        variant("custom") {
+            // configuring report tasks for custom build variant
+        }
+    }
+}
+```
+
+To configure of the report generation for all classes of merged modules, it is necessary to specify the following in the build script file of the merging module (and only in it):
+```kotlin
+kover {
+    reports {
+        total {
+            // configuring report tasks
+        }
+    }
+}
+```
+
+Available configuration options are listed in a [separate section](#configuring-report-tasks).
+
+----
+
+### Kotlin multiplatform mixed Android and JVM project
+#### Application of Kover plugin
+Add the following to build file in each module of your Gradle build:
+
+```kotlin
+plugins {
+     id("org.jetbrains.kotlinx.kover") version "0.7.6"
+}
+```
+It is recommended to apply the Kover Plugin in the root module, even if there is no source code or tests there.
+You [do not need to specify](#application-in-submodule-with-plugins-dsl) the plugin version in all submodules.
+
+For more information about application of the plugin, refer to the [relevant section](#apply-kover-gradle-plugin-in-project)
+
+After applying Kover Gradle plugin, during the running tests, the classes are modified (instrumented) when loaded into the JVM which may lead to some performance degradation, or affect concurrent tests.
+In order to learn more about instrumentation, and it's configuring, please refer to the [corresponded section](#instrumentation).
+
+#### Create custom Report Variants
+The module with Android Gradle Plugin or Kotlin multiplatform with android target is divided into different **build variants** (flavor + build type), which may have different source sets and tests.
+For each Gradle Build Variant, Kover creates its own **report variant** that have the same names.
+Such **report variant** includes only the classes declared in the corresponding **build variant**, and also runs only tests this **build variant**.
+
+For JVM target in Kotlin multiplatform module or for Kotlin JVM module Kover creates special **report variant** with name `jvm`. 
+Reports for `jvm` **report variant** contains classes and JVM tests from JVM target or all classes from Kotlin JVM module. 
+
+If there is a need to generate report from classes of several **build variants**, then you could create custom **report variant**:
+```kotlin
+kover {
+    currentProject {
```

Comment:
`currentProject` is not explained anywhere. What happens if I forgot to add it? Is it required only for merging project (which is explained below, not above)?

Reply:
This is needed only if it is necessary to put classes from several Android Build Variants in one report at once.

> What happens if I forgot to add it?
Then the `custom` report variant will not be created and you will not be able to call the `koverHtmlReportCustom` task :).

Reply:
Ah, `createVariant` is defined in it. Why can't it be just `kover { createVariant(...) {}}` though?

Reply:
In this case we will need to move all the functions of working with variants there: `createVariant`, `providedVariant`, `totalVariant` - it seemed to me that it would be too much